pkgs/development/python-modules/openai-triton/default.nix

```diff
@@ -0,0 +1,235 @@
+{ lib
+, buildPythonPackage
+, python
+, fetchurl
+, fetchFromGitHub
+, addOpenGLRunpath
+, cmake
+, cudaPackages ? { }
+, llvmPackages
+, pybind11
+, gtest
+, zlib
+, ncurses
+, libxml2
+, lit
+, filelock
+, torchWithRocm
+, pytest
+, pythonRelaxDepsHook
+}:
+
+let
+  pname = "triton";
+  version = "2.0.0";
+
+  inherit (cudaPackages) cuda_nvcc cuda_cudart backendStdenv;
+  ptxas = "${cuda_nvcc}/bin/ptxas";
+in
+buildPythonPackage {
+  inherit pname version;
+
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "openai";
+    repo = "triton";
+    rev = "v${version}";
+    hash = "sha256-9GZzugab+Pdt74Dj6zjlEzjj4BcJ69rzMJmqcVMxsKU=";
+  };
+
+  patches = [
+    # Prerequisite for llvm15 patch
+    (fetchurl {
+      url = "https://github.com/openai/triton/commit/2aba985daaa70234823ea8f1161da938477d3e02.patch";
+      hash = "sha256-HEuLZFif++a/fKs3dyIhqSc+D2DPbzEXOSSR4nRWtgQ=";
+    })
+    (fetchurl {
+      url = "https://github.com/openai/triton/commit/e3941f9d09cdd31529ba4a41018cfc0096aafea6.patch";
+      hash = "sha256-sl8woykSLCq4ZJYzEQdCPWM8rzv+s4RTK7PuqsTcmy8=";
+    })
+
+    # Not using fetchurl because needed to remove binary diff for ptxas
+    ./llvm15.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace python/setup.py \
+      --replace \
+        '= get_thirdparty_packages(triton_cache_path)' \
+        '= os.environ["cmakeFlags"].split()'
+  ''
+  # A typo in setup.py?
+  # NOTE: Ok, now I'm disabling setuptools check anyway
+  # + ''
+  #   substituteInPlace python/setup.py \
+  #     --replace '"tests"' '"test"'
+  # ''
+  # Wiring triton=2.0.0 with llcmPackages_rocm.llvm=5.4.3
+  # Revisit when updating either triton or llvm
+  + ''
+    substituteInPlace CMakeLists.txt \
+      --replace "nvptx" "NVPTX" \
+      --replace "LLVM 11" "LLVM"
+    sed -i '/AddMLIR/a set(MLIR_TABLEGEN_EXE "${llvmPackages.mlir}/bin/mlir-tblgen")' CMakeLists.txt
+    sed -i '/AddMLIR/a set(MLIR_INCLUDE_DIR ''${MLIR_INCLUDE_DIRS})' CMakeLists.txt
+    find -iname '*.td' -exec \
+      sed -i \
+      -e '\|include "mlir/IR/OpBase.td"|a include "mlir/IR/AttrTypeBase.td"' \
```

Comment:
These paths and class names seem to change a lot between releases, so, admittedly, the next update is probably going to be involved